### PR TITLE
added support for Moes Radiator Thermostat  TZE200_fhn3negr

### PR DESCRIPTION
**Pull request by wollo**
see issue 1803

### CODE REVIEW

File: converters/toZigbee.js [hunk 1]
```diff
@@ -2041,7 +2041,182 @@ const converters = {
             }
         },
     },
-    tuya_led_control: {
+    moes_thermostat_current_heating_setpoint: {
```
Comment (Koenkk) [inline converters/toZigbee.js:2044]: Same as https://github.com/Koenkk/zigbee-herdsman-converters/blob/0a6809fa49cf5530956f2b48b01aa4d993013141/converters/toZigbee.js#L1980

File: converters/toZigbee.js [hunk 2]
```diff
@@ -2041,7 +2041,182 @@ const converters = {
...
+        key: ['current_heating_setpoint'],
+        convertSet: async (entity, key, value, meta) => {
+            await tuya.sendDataPointValue(entity, tuya.dataPoints.moesHeatingSetpoint, value);
+        },
+    },
+    moes_thermostat_min_temperature: {
```
Comment (Koenkk) [inline converters/toZigbee.js:2050]: Same as https://github.com/Koenkk/zigbee-herdsman-converters/blob/0a6809fa49cf5530956f2b48b01aa4d993013141/converters/toZigbee.js#L1986

File: converters/toZigbee.js [hunk 3]
```diff
@@ -2041,7 +2041,182 @@ const converters = {
...
+        key: ['min_temperature'],
+        convertSet: async (entity, key, value, meta) => {
+            await tuya.sendDataPointValue(entity, tuya.dataPoints.moesMinTemp, value);
+        },
+    },
+    moes_thermostat_calibration: {
```
Comment (Koenkk) [inline converters/toZigbee.js:2056]: Same as https://github.com/Koenkk/zigbee-herdsman-converters/blob/0a6809fa49cf5530956f2b48b01aa4d993013141/converters/toZigbee.js#L1992

File: converters/toZigbee.js [hunk 4]
```diff
@@ -2041,7 +2041,182 @@ const converters = {
...
+        key: ['local_temperature_calibration'],
+        convertSet: async (entity, key, value, meta) => {
+            if (value < 0) value = 4096 + value;
+            await tuya.sendDataPointValue(entity, tuya.dataPoints.moesTempCalibration, value);
+        },
+    },
+    moes_thermostat_mode: {
```
Comment (Koenkk) [inline converters/toZigbee.js:2063]: Same as https://github.com/Koenkk/zigbee-herdsman-converters/blob/0a6809fa49cf5530956f2b48b01aa4d993013141/converters/toZigbee.js#L1999

**Comment by Koenkk** [comment]
Can you rebase this PR against master? It seems there is a lot of code added under https://github.com/Koenkk/zigbee-herdsman-converters/pull/2209/files which is already there.

File: converters/toZigbee.js [hunk 5]
```diff
@@ -2041,7 +2041,182 @@ const converters = {
...
+        },
+    },
+    moes_alt_thermostat_current_heating_setpoint: {
+        key: ['current_heating_setpoint'],
+        convertSet: async (entity, key, value, meta) => {
+            const temp = Math.round(value * 2);
+            if (meta.state.system_mode == 'heat') {
+                await tuya.sendDataPointValue(entity, tuya.dataPoints.moesHeatingSetpoint, temp);
+            } else if (meta.state.system_mode == 'auto') {
+                await tuya.senDataPointValue(entity, tuya.dataPoints.moesAltScheduleTempOverride, temp);
```
Comment (basveeling) [inline converters/toZigbee.js:2101]: Thanks for making a PR for this device wollo! I will try and test things where I can. I spotted this typo for now:
`send*DataPoint`

File: devices.js [hunk 6]
```diff
@@ -14125,7 +14125,51 @@ const devices = [
             .withSystemMode(['off', 'heat', 'auto'], ea.STATE_SET).withLocalTemperatureCalibration(ea.STATE_SET)
             .withRunningState(['idle', 'heat'], ea.STATE).withAwayMode()],
     },
+    {
+        // Moes Tuya Alt Thermostat
+        fingerprint: [
+            {modelID: 'TS0601', manufacturerName: '_TZE200_fhn3negr'},
+            { modelID: 'TS0601', manufacturerName: '_TZE200_i48qyn9s' },
```
Comment (basveeling) [inline devices.js:14132]: I believe my manufacturerName's include:
```
        { modelID: 'TS0601', manufacturerName: '_TZE200_zion52ef' },
        { modelID: 'TS0601', manufacturerName: '_TZE200_qc4fpmcn' },
```

**Pull request closed without merging**